helm/loki/values.schema.json

```diff
@@ -0,0 +1,1559 @@
+                            },
+                            "additionalProperties": false
+                        },
+                        "requests": {
+                            "type": "object",
+                            "properties": {
+                            "cpu": { "type": "string" },
+                            "memory": { "type": "string" }
+                            },
+                            "additionalProperties": false
+                        }
+                    },
+                    "additionalProperties": false
+                },
+                "serviceAccount": {
+                    "type": "object",
+                    "properties": {
+                        "annotations": {
+                            "type": "object"
+                        },
+                        "create": {
+                            "type": "boolean"
+                        },
+                        "imagePullSecrets": {
+                            "type": "array"
+                        },
+                        "name": {
+                            "type": ["string", "null"]
+                        }
+                    }
+                },
+                "terminationGracePeriodSeconds": {
+                    "type": "integer"
+                },
+                "tolerations": {
+                    "type": "array"
+                }
+            }
+        },
+        "distributor": {
+            "type": "object",
+            "properties": {
+                "affinity": {
+                    "type": "string"
+                },
+                "extraArgs": {
+                    "type": "array"
+                },
+                "extraEnv": {
+                    "type": "array"
+                },
+                "extraEnvFrom": {
+                    "type": "array"
+                },
+                "extraVolumeMounts": {
+                    "type": "array"
+                },
+                "extraVolumes": {
+                    "type": "array"
+                },
+                "image": {
+                    "type": "object",
+                    "properties": {
+                        "registry": {
+                            "type": ["string", "null"]
+                        },
+                        "repository": {
+                            "type": ["string", "null"]
+                        },
+                        "tag": {
+                            "type": ["string", "null"]
+                        }
+                    }
+                },
+                "nodeSelector": {
+                    "type": "object"
+                },
+                "podAnnotations": {
+                    "type": "object"
+                },
+                "priorityClassName": {
+                    "type": ["string", "null"]
+                },
+                "replicas": {
+                    "type": "integer",
+                    "minimum": 3
```

Comment:
prevent less than three replicas

Reply:
Can this also run with one replica? I would set the defaults to three in the values file. But there is always a customer or maybe even us to run with a smaller footprint just for testing. Strict restriction lead to requests for exceptions :)

Reply:
I'm not sure, I think it could, but the loki docs say 
> If an ingester process crashes or exits abruptly, all the data that has not yet been flushed will be lost. Loki is usually configured to replicate multiple replicas (usually 3) of each log to mitigate this risk.

So now should be allow customers (and us) to run potential unsafe configurations?

Reply:
Discussed this in person, I'll remove the strict validation here and add a note to the readme